### DISCUSSION TITLE
How to move a web page built with Cassiopea template to a new one using Astroid???

### DISCUSSION THREAD

**Discussion by AntonioSco** [templaza/astroid-framework#896] [Q&A] [answered]
Hi, is there any tutorial showing how to transform a webpage built using the cassiopea template (or any other template) to a new one using the Astroid template?
I reading through the only official document I found on the Astroid website and to the list of videos I found in youtube but they seem to address specific aspects of Astroid but could not found any that explains how to build a page from scratch where the page uses say
a header at th every top of a page for the contact details
a header that shows the logo + menu + search
a header showing an image (static or animated)
Etc. that is how to create to structure of the webpage form scratch and include and or distribute all the already existing modules

The other thing is that, I make changes in the Astroid configuration panel, but the changes are not displayed neither in the preview nor in the page re-loaded itself....don't understand why and this makes it difficult for me to use the framework. Has anyone experienced this as well?
Any help appreciated

**Comment by paurray** [comment]
Hi @AntonioSco 

you could try this playlist that I created here:

Your 1st Bare Bones Website with Astroid & Joomla
https://www.youtube.com/playlist?list=PLOd7DxRu2yaJvVJKPEGZAOGoWpWOyNL1b

Please check it out and give me feed back as I am looking at upping my game and creating more content like this in 2025!

thanks

Paul

**Reply by AntonioSco** [reply]
Thanks mate! I'll check it out once I make the framework work!

**Comment by paurray** [comment]
@AntonioSco 
Also could you double check what exactly you installed. Should look something like this with Templates and Framework...

<img width="960" alt="astroidFullInstall" src="https://github.com/user-attachments/assets/37d064b1-16fe-46d2-b986-f2b77d44abb3">

**Reply by AntonioSco** [reply]
Hi @paurray ,
thanks for your note! I just checked and guess what? The **Astroid Package** is not there!!! How is this possible? I installed from the astroid-framework-3.1.8.zip file downloaded from the [Astroid website](https://astroidframe.work/)....How/where can I get the complete extension?

**Reply by paurray** [reply]
Hi @AntonioSco 
I just tried the Astroid Framework 3.1.8.zip file and it worked for me.
https://github.com/templaza/astroid-framework/releases

I had to first install the framework: System > Install > Extensions
Then after install I navigated to System > Templates > Site Template Styles and saw this

<img width="960" alt="Astroid-Install01" src="https://github.com/user-attachments/assets/d68ff227-ee2e-4e3d-9baa-93b8faf1a52f">

Then I selected Two and saw this

<img width="960" alt="Astroid-Install02" src="https://github.com/user-attachments/assets/fd773484-b503-400d-a6d2-879455d02eee">

Then Saw this

<img width="960" alt="Astroid-Install03" src="https://github.com/user-attachments/assets/531ac92e-2a1c-4894-ad67-f8398dbd7c4a">

This is kind of new and not covered in my tutorials currently :-(
I suggest trying again (I am assuming that you are NOT working on a live site rather a development site)

**Reply by AntonioSco** [reply]
yes I am not working on  a live site

> @AntonioSco Also could you double check what exactly you installed. Should look something like this with Templates and Framework...
> <img alt="astroidFullInstall" width="960" src="https://private-user-images.githubusercontent.com/9163424/391434474-37d064b1-16fe-46d2-b986-f2b77d44abb3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NzM4NDUsIm5iZiI6MTczMzc3MzU0NSwicGF0aCI6Ii85MTYzNDI0LzM5MTQzNDQ3NC0zN2QwNjRiMS0xNmZlLTQ2ZDItYjk4Ni1mMmI3N2Q0NGFiYjMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMTk0NTQ1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MWY5MzgwODI1ZDE4MTg4ZjczYzRlZTY2ZGM0ZWU4ZDVmNDBiMzExYzI5NDk2MDdmYWVmMzk3ZTdhNWFkNzI2MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.PlziOQ7q75gxVdX-tFff9iGaeTZA28PcZWVcb2v-REo">

HI @paurray,
again Is the "Astroid Package" that appears in your listing a component of the current release of Astroid that **needs** to be in place for Astroid to funcion properly? Because I reinstalled Astroid twice already and I never find such a component installed.

**Reply by paurray** [reply]
@AntonioSco The theory is that after installing Astroid Framework 3.1.8.zip and navigating to System > Templates > Site Template Styles, you should see a prompt to install one of three templates to use the framework. You can access the Astroid Framework via Templates once at least one template is successfully installed. However, something likely went wrong initially, and it seems the Astroid Framework thinks templates are installed when they are not.

It might be worth exploring the option to install a template from an earlier release, though I’m not sure if that’s advisable. A clean install could be the best approach to resolve the issue and try again. My 2 Cents.

**Comment by Banuk1616** [comment]
@AntonioSco 

I think the simple conversion of a template, for example a template based on Cassiopia or others with a few clicks will not be possible. 

But if you understand the Astroid Framework and the Layout Manager, then it is relatively easy to rebuild a template that was not created with the Astroid Framework. 

But this requires a certain degree of familiarisation with the Astroid Framework and some knowledge of CSS. The nice thing about the Astroid Framework is that it's fun to work with and learn CSS and SCSS at the same time. It goes hand in hand and almost by itself, you just have to take your time.


The problems you describe are not familiar to me and I have not been able to find a thread with similar problems here in the forum.

What might help is if you download and install the Quickstart. However, you must ensure that Joomla is already included in the Quickstart and that it cannot be installed in an existing Joomla installation.

In the Quickstart you can see how most of the functions in Astroid Famework work. It is really much easier than it seems.

As Joomla 4 is coming to an end, I would also upgrade everything to Joomla 5.

My English is not that good, but if it is possible despite the language barrier, I will be happy to help you.

Translated with DeepL.com (free version)

**Reply by AntonioSco** [reply]
Hi @Banuk1616 
thanks for you note and your willingness to help out!
Thanks for all your suggestions, I'll check them out...DeepL is doing a good job after all :-)

**Comment by AntonioSco** [comment]
Hi @Banuk1616 why do you suggest to using quickstart? I already have a joomla 4.4.9 
installation which is working.

**Reply by Banuk1616** [reply]
I had recommended that you take a look at the Quickstart so that you can see how Astroid works and how relatively easy it is to rebuild a website created with Cassiopeia or another template in a very similar way with Astroid. Your initial question was how to convert a website created with Cassiopeia or another template into a website that works with Astroid. As far as I know, a simple conversion to Astroid is not possible. 
Best Regards

**Comment by AntonioSco** [comment, marked as answer]
Hi all, I finally figured out what the problem was. When I first started using Cassiopeia template I did something whose repercussions didn't know about. Basically I assigned every menu element to the cassiopeia template in all of the pages of my website. So even if I assigned a different template as a default template for my site, it would always show the pages styled according to cassiopeia, no matter what other template I'd use.
Yesterday night though, I noticed something which I hadn't noticed before in the templates panel, that is that cassiopeia had the string "Assigned on 14 pages" in the **Pages** column, which intrigued me;
![assign](https://github.com/user-attachments/assets/3f3757eb-80e8-48d8-8c83-8967496d955a)

So I proceeded to change that (and learn what it meant!) and...voila! Right after doing this, the Astroid template was finally fully operational.
Anyway thanks to all those who have tried to help me!
Best regards
